lib/apiservers/engine/backends/container.go

```diff
@@ -189,75 +188,18 @@ const (
 	defaultEnvPath = "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
 )
 
+// TODO: we should probably just make this purely a proxy thing. This is now just a passthrough
 func (c *Container) Handle(id, name string) (string, error) {
-	resp, err := c.containerProxy.Client().Containers.Get(containers.NewGetParamsWithContext(ctx).WithID(id))
-	if err != nil {
-		switch err := err.(type) {
-		case *containers.GetNotFound:
-			cache.ContainerCache().DeleteContainer(id)
-			return "", NotFoundError(name)
-		case *containers.GetDefault:
-			return "", InternalServerError(err.Payload.Message)
-		default:
-			return "", InternalServerError(err.Error())
-		}
-	}
-	return resp.Payload, nil
+	return c.containerProxy.Handle(id, name)
```

Comment:
I've got a strong preference *not* to move the cache management (the NotFound case) into proxy - it's supposed to be purely for the client management, marshalling, unmarshalling, and handling of error type translation.
The cache is a specific behaviour that the personality layers on top so should remain out of the proxy.

That said I'm not sure this needs to be exported from Container (captial H) - I don't think this is implementing an interface that has this as part of it's signature. May be wrong.

Reply:
It is not, but it does make a call via swagger which is why it has been moved there. I can pull the cache portion out, but  I still think the `GetHandle` style endpoint should be move to the proxy. I kept that function as a passthrough since it is used in several other functions and I was making an effort to not add additional bulk to this PR.

Reply:
Upon rereading this with some sleep, I think that is exactly what you were asking for. :) I will be moving the cache handling of the not found case back into the container.go file rather than dumping the entire function handling into the proxy.